lib/elixir/lib/exception.ex

```diff
@@ -949,28 +951,161 @@ defmodule MismatchedDelimiterError do
 
   @impl true
   def message(%{
-        line: _start_line,
-        column: _start_column,
+        line: start_line,
+        column: start_column,
         end_line: end_line,
         end_column: end_column,
         description: description,
         file: file,
         snippet: snippet
       }) do
-    snippet =
-      :elixir_errors.format_snippet(
-        {end_line, end_column},
-        file,
-        description,
-        snippet,
-        :error,
-        [],
-        nil
-      )
+    start_pos = {start_line, start_column}
+    end_pos = {end_line, end_column}
+    lines = String.split(snippet, "\n")
 
+    snippet = format_snippet(start_pos, end_pos, description, file, lines)
     format_message(file, end_line, end_column, snippet)
   end
 
+  defp format_snippet(
+         {start_line, start_column} = start_pos,
+         {end_line, end_column} = end_pos,
+         description,
+         file,
+         lines
+       )
+       when start_line < end_line do
+    max_digits = digits(end_line)
+    general_padding = max(2, max_digits) + 1
+    padding = n_spaces(general_padding)
+
+    relevant_lines =
+      if end_line - start_line < @max_lines_shown do
+        line_range(lines, start_pos, end_pos, padding, max_digits)
+      else
+        first_line = Enum.at(lines, start_line - 1)
+        last_line = Enum.at(lines, end_line - 1)
```

Comment:
It seems we assume those lines indeed exist, right? So it is best to avoid nils propagating:

```suggestion
        first_line = Enum.fetch!(lines, start_line - 1)
        last_line = Enum.fetch!(lines, end_line - 1)
```